letter-sender/src/main/java/com/christmas/letter/sender/dto/ChristmasLetter.java

```diff
@@ -0,0 +1,32 @@
+package com.christmas.letter.sender.dto;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class ChristmasLetter {
+
+    @NotNull(message = "email.cannot.be.null")
+    @Email(regexp = "^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^.-]+@[a-zA-Z0-9.-]+$", message = "email.must.be.valid")
+    private String email;
+
+    @NotNull(message = "name.cannot.be.null")
+    @NotEmpty(message = "name.cannot.be.empty")
```

Comment:
If you only keep the @NotEmpty annotation, doesn't it cover also null values? Same for other fields.

Reply:
yes technically it does, just wanted to give a more specific error regarding the request, so that the client doing the request will know if the field is just empty or missing completely from request body. But will update to keep just the @notempty annotation :)